src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table-actions/table-actions.component.spec.ts

```diff
@@ -17,18 +17,14 @@ describe('TableActionsComponent', () => {
   let unprotectAction: CdTableAction;
   let deleteAction: CdTableAction;
   let copyAction: CdTableAction;
+  let falseAction: CdTableAction;
+  let trueAction: CdTableAction;
+  let actions: CdTableAction[];
   let scenario;
   let permissionHelper: PermissionHelper;
 
   const setUpTableActions = () => {
-    component.tableActions = [
-      addAction,
-      editAction,
-      protectAction,
-      unprotectAction,
-      copyAction,
-      deleteAction
-    ];
+    component.tableActions = [...actions];
```

Comment:
I'm rather new to TS and now sure why this is done this way. Can you please explain to me why the destructuring of `actions` is necessary here? And wouldn't it be better to simply pass `actions` as an argument to `setupTableActions`?